src/string/grapheme/grapheme.cr

```diff
@@ -0,0 +1,258 @@
+require "./properties"
+
+class String
+  # Iterates the grapheme boundaries in this string and yields for each grapheme
+  # cluster the byte index of the first character and the first character of the
+  # cluster, as well as the last character of the grapheme (which can be used
+  # to avoid multiple decoding in single-char graphemes).
+  private def each_grapheme_boundary(& : Range(Int32, Int32), Char -> Nil) : Nil
```

Comment:
I'm not sure if I'm getting right the comment: I'm expecting it from the description to yield an index and two Chars.

Reply:
Ah yeah, the second `first character` must be `last character`. It returns the byte indices of the first and last characters.